packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1021,6 +1021,15 @@ class _WidgetsAppState extends State<WidgetsApp> implements WidgetsBindingObserv
     });
   }
 
+  // RENDERING
+  @override
+  void didChangePlatformBrightness() {
+    setState(() {
+      // The platformBrightness property of ui.window has changed. We reference
```

Comment:
I think this will work but it's a bit too coincidental. The actual references and usages are in MaterialApp which won't rebuild.

Reply:
This shouldn't be necessary at all. MaterialApp's theme Builder should rebuild when the brightness changes because MediaQuery.of()

Reply:
I removed the setState() as per @HansMuller's comment